File: ur_typical90/typical90_o.py
class Combination():
    def __init__(self, N, MOD):
        self.mod = MOD
        self.FACT = [1, 1]
        self.INV = [0, 1]
        self.FACTINV = [1, 1]
        for i in range(2, N + 1):
            self.FACT.append((self.FACT[-1] * i) % self.mod)
            self.INV.append(pow(i, self.mod - 2, self.mod))
            self.FACTINV.append((self.FACTINV[-1] * self.INV[-1]) % self.mod)

    def count(self, N, R):
        if (R < 0) or (N < R):
            return 0
        R = min(R, N - R)
        return self.FACT[N] * self.FACTINV[R] * self.FACTINV[N-R] % self.mod


n = int(input())
MOD1 = 10**9 + 7

cmb = Combination(n, MOD1)
for i in range(1, n+1):
    ans, sup = 0, (n-1) // i + 1
    for j in range(sup):
        ans += cmb.count(n - i*j + j, j+1)
    print(ans % MOD1)

File: ur_typical90/typical90_o.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

const ll MOD1 = 1e9 + 7;

#define rep(i, n) for (ll i = 0; i < ll(n); i++)
#define repi(i, a, b) for (ll i = ll(a); i < ll(b); i++)

ll mod_pow(ll x, ll y, const ll& mod) {
    ll res = 1;
    while (y > 0) {
        if (y & 1) res = (res * x) % mod;
        x = (x * x) % mod;
        y >>= 1;
    }
    return res;
}

class Combination {
   private:
    vector<ll> fact, inv, factinv;

   public:
    ll N, MOD;

    Combination(ll n0, ll mod) {
        N = n0, MOD = mod;
        fact = vector<ll>(N + 1, -1);
        inv = vector<ll>(N + 1, -1);
        factinv = vector<ll>(N + 1, -1);

        fact[0] = 1, fact[1] = 1;
        inv[0] = 0, inv[1] = 1;
        factinv[0] = 1, factinv[1] = 1;
        repi(i, 2, N + 1) {
            fact[i] = (fact[i - 1] * i) % MOD;
            inv[i] = mod_pow(i, MOD - 2, MOD);
            factinv[i] = (factinv[i - 1] * inv[i]) % MOD;
        }
    }

    ll count(ll cn, ll cr) {
        if ((cr < 0) || (cn < cr)) return 0;
        cr = min(cr, cn - cr);
        ll res = fact[cn] * factinv[cr] % MOD;
        res = res * factinv[cn - cr] % MOD;
        return (res % MOD + MOD) % MOD;
    }
};

int main() {
    ll n;
    cin >> n;

    Combination cmb(n, MOD1);
    repi(i, 1, n + 1) {
        ll ans = 0, sup = (n - 1) / i + 1;
        rep(j, sup) ans += cmb.count(n - i * j + j, j + 1);
        cout << ans % MOD1 << '\n';
    }
    return 0;
}
